packages/@webex/plugin-cc/test/unit/spec/services/config/index.ts

```diff
@@ -243,10 +260,580 @@ describe('AgentConfigService', () => {
+        tenantId: 'tenant123',
+        timezone: 'GMT',
+      };
+
+      const mockOrgSettings = {
+        campaignManagerEnabled: true,
+        webRtcEnabled: true,
+        maskSensitiveData: false,
+      };
+
+      const mockTenantData = {
+        timeoutDesktopInactivityEnabled: false,
+        timeoutDesktopInactivityMins: 15,
+        forceDefaultDn: true,
+        dnDefaultRegex: 'regexUS',
+        dnOtherRegex: 'regexOther',
+        privacyShieldVisible: true,
+        outdialEnabled: true,
+        endCallEnabled: true,
+        endConsultEnabled: true,
+        callVariablesSuppressed: false,
+      };
+
+      const mockURLMapping = [
+        {key: 'ACQUEON_API_URL', url: 'https://api.example.com'},
+        {key: 'ACQUEON_CONSOLE_URL', url: 'https://console.example.com'},
+      ];
+
+      const mockAuxCodes = [
+        {id: 'aux1', type: 'WRAP_UP_CODE', name: 'Wrap Up Code 1', isDefault: true},
+        {id: 'aux2', type: 'IDLE_CODE', name: 'Idle Code 1', isDefault: true},
+      ];
+
+      const parseAgentConfigsSpy = jest.spyOn(util, 'parseAgentConfigs');
+      agentConfigService.getUserUsingCI = jest.fn().mockResolvedValue(mockUserConfig);
+      agentConfigService.getOrgInfo = jest.fn().mockResolvedValue(mockOrgInfo);
+      agentConfigService.getOrganizationSetting = jest.fn().mockResolvedValue(mockOrgSettings);
+      agentConfigService.getTenantData = jest.fn().mockResolvedValue(mockTenantData);
+      agentConfigService.getURLMapping = jest.fn().mockResolvedValue(mockURLMapping);
+      agentConfigService.getAllAuxCodes = jest.fn().mockResolvedValue(mockAuxCodes);
+      agentConfigService.getDesktopProfileById = jest.fn().mockResolvedValue(mockAgentProfile);
+      agentConfigService.getDialPlanData = jest.fn().mockResolvedValue(mockDialPlanData);
+      agentConfigService.getAllTeams = jest.fn().mockResolvedValue(mockTeamData);
+
+      const result = await agentConfigService.getAgentConfig(mockOrgId, mockAgentId);
+
+      expect(LoggerProxy.logger.info).toHaveBeenCalledWith('Fetched user data');
+      expect(LoggerProxy.logger.info).toHaveBeenCalledWith('Fetched all required data');
+      expect(LoggerProxy.logger.info).toHaveBeenCalledWith('Parsing completed for agent-config');
+      expect(LoggerProxy.logger.info).toHaveBeenCalledWith(
+        'Fetched configuration data successfully'
+      );
+      expect(parseAgentConfigsSpy).toHaveBeenCalledTimes(1);
+
+      expect(parseAgentConfigsSpy).toHaveBeenCalledWith({
+        userData: mockUserConfig,
+        teamData: mockTeamData,
+        tenantData: mockTenantData,
+        orgInfoData: mockOrgInfo,
+        auxCodes: mockAuxCodes,
+        orgSettingsData: mockOrgSettings,
+        agentProfileData: mockAgentProfile,
+        dialPlanData: mockDialPlanData,
+        urlMapping: mockURLMapping,
+      });
+    });
+
+    it('should fetch and parse agent configuration with different values and conditions successfully', async () => {
```

Comment:
Is there a test that contains all the properties inside the agent object? These tests seem to contain only a subset?

Reply:
yes. first test (above this test) has all the values. this test has sub set of values leftover conditions.